AI/Nullkiller/Pathfinding/AINodeStorage.cpp

```diff
@@ -69,23 +69,25 @@ void AINodeStorage::initialize(const PathfinderOptions & options, const CGameSta
 
 	//TODO: fix this code duplication with NodeStorage::initialize, problem is to keep `resetTile` inline
 	const PlayerColor fowPlayer = ai->playerID;
-	const auto & fow = static_cast<const CGameInfoCallback *>(gs)->getPlayerTeam(fowPlayer)->fogOfWarMap;
+	const auto fow = static_cast<const CGameInfoCallback *>(gs)->getPlayerTeam(fowPlayer)->fogOfWarMap;
 	const int3 sizes = gs->getMapSize();
 
-	parallel_for(blocked_range<size_t>(0, sizes.x), [&](const blocked_range<size_t>& r)
+	for(int z = 0; z < sizes.z; ++z) //Can't parallelize only 2 layers
 	{
```

Comment:
We still can put it first loop before for(pos.x = r.begin()...

Reply:
Oh, it works indeed :)